### PR TITLE
Yujin release

### PR DESCRIPTION
**Pull request by asmodehn**
small script to do a bloom release versus our own rosdistro.
@stonier please have a look and let me know if we need anything more/less.

### CODE REVIEW

**Comment by stonier** [comment]
Do you want that script actually getting released in the deb? It'll need to be put in the `setup.py` as well.

Actually those scripts...i expect that git module is bringing in mike's script alongside it.

**Comment by asmodehn** [comment]
I added them to the setup.py, but this is a bit dirty...

We depend on rosdep-generator, so I d like to bring it in as a dependency or something, instead of a submodule actually... but so far it looks like the dependencies are not specified in our package... or did I miss something ? But rosdep-generator is not a package anyway...

The submodule forces it to be in a subdirectory, and since we install everything at the same level the yujin_release script will now need the rosdep-generator installed to work (even when you run from source). Works but could be better...
Any simple idea to fix this ?

**Comment by stonier** [comment]
I never run my scripts from source in this package since it's always important to me to check that the installed versions work (which is what everyone uses). I typipcally hack the script in the installed location and then copy it back here, commit, install and test again.

If you do want to run it from source, would a symlink from `scripts/rosdep-generator` to `scripts/rosdep-generator/rosdep-generator` do?

Not much we can do about depending on it if mike hasn't got a conveniently available deb for it. He might...in some clearpath apt repository or something. But I don't want to add a clearpath repository to our list for just one script.

**Comment by stonier** [comment]
If you do find it a long term requirement for us, maybe you can get him to put it into the rosdep package or whatever is appropriate. I wouldn't go about creating noise and effort this early yet though.

**Comment by asmodehn** [comment]
I thought about the symlink but didnt know if it would be more useful or annoying. I think we can keep it just like that then, since our usage is similar.

Maybe when we have time to make nice clean pip packages with proper dependencies and so on, we can improve this... but that will probably be after we find a good solution for that rosdep-generator code anyway.

File: setup.py [hunk 1]
```diff
@@ -35,6 +35,8 @@
            'scripts/yujin_tools_settings',
            'scripts/yujin_list_dirs_by_size',
            'scripts/yujin_list_git_branches',
+           'scripts/rosdep-generator/rosdep-generator',
```
Comment (stonier) [inline setup.py]: This one will be pointing to the wrong place...`scripts/rosdep-generator` I believe?

Comment (asmodehn) [inline setup.py]: I dont think so... It should get the file from inside the dir, and copy it in the same place as other scripts.

File: scripts/yujin_release [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+DISTRO=$1
+#package name
+PKG=$2
+#useful for --edit for example
+EXTRA=$3
+
+#Import ROS rosdep to be able to resolve public dependencies
+# AlexV - this is still magic for me
+PWD=`pwd`
+rosdep-generator --distro indigo
```
Comment (stonier) [inline scripts/yujin_release]: Should be the DISTRO environment variable right?

File: setup.py (hunk 1 above)
Comment (stonier) [inline setup.py]: Ah, right...I missed something there.

File: scripts/yujin_release [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+echo "Executing next steps..."
+set -x
+sudo sh -c "echo \"yaml file://$PWD/ros-rosdistro-indigo.yaml indigo\" > /etc/ros/rosdep/sources.list.d/90-ros-rosdistro-$DISTRO.list"
+set +x
+rosdep update
+echo "Done."
+
+# Specifying rosdistro
+export ROSDISTRO_INDEX_URL=https://raw.githubusercontent.com/yujinrobot/rosdistro/master/index.yaml
```
Comment (stonier) [inline scripts/yujin_release]: Ah, this baby lets us point the pull request submission to our repo?

**Comment by asmodehn** [comment]
Now using $DISTRO everywhere, I think. And replaced submodule, so this should be good to merge...

**Comment by stonier** [comment]
Released in version 0.3.1.

**Pull request merged**